includes/suppress-ads.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * Controls for suppressing ads on certain singles.
+ *
+ * @package Newspack
+ */
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Register meta for suppressing ads on singles.
+ */
+function newspack_register_suppress_ad_meta() {
+	register_post_meta(
+		'',
+		'newspack_suppress_ads',
+		array(
+			'show_in_rest' => true,
+			'single'       => true,
+			'type'         => 'boolean',
+		)
+	);
+}
+add_action( 'init', 'newspack_register_suppress_ad_meta' );
+
+/**
+ * Enqueue editor assets for suppressing ads.
+ */
+function newspack_enqueue_suppress_ad_assets() {
+	if ( 'post' === get_current_screen()->post_type || 'page' === get_current_screen()->post_type ) {
+		wp_enqueue_script( 'newspack-suppress-ads', Newspack_Ads::plugin_url( 'dist/suppress-ads.js' ), [], NEWSPACK_ADS_VERSION, true );
+	}
+}
+add_action( 'enqueue_block_editor_assets', 'newspack_enqueue_suppress_ad_assets' );
+
+/**
+ * Get whether ads should be displayed on a screen.
+ *
+ * @param int $post_id Post ID to check (optional, default: current post).
+ * @return bool
+ */
+function newspack_should_show_ads( $post_id = null ) {
```

Comment:
```suggestion
function newspack_ads_should_show_ads( $post_id = null ) {
```

Should probably prefix with the full name of the plugin.

Reply:
That makes sense. Good suggestion. https://github.com/Automattic/newspack-ads/pull/26/commits/103be8c309059f4806d9a4b4ac5f5d080649bfef#diff-7af73f10e634570be7ed9d3b5db857d2R13